src/main/java/org/gwtbootstrap3/extras/toggleswitch/client/ui/base/ToggleSwitchBase.java

```diff
@@ -206,21 +209,59 @@ public void setValue(final Boolean value) {
         setValue(value, false);
     }
 
+    /**
+     * Called when this widget is attached to the DOM.
+     * 
+     * Here we want to make sure the state of the toggle switch hasn't changed before it was 
+     * attached or re-attached to the DOM.  If the value has changed update the toggle switch 
+     * via {@link #switchState(Element, boolean, boolean)}.
+     */
+    @Override
+    public void onAttach() {
+        super.onAttach();
+        boolean switchValue = switchState(getElement());
+        boolean value = getValue();
+        if (value != switchValue) {
+            switchState(getElement(), value, true);
+        }
+    }
+    
+    /**
+     * Sets the value of this widget.
+     * 
+     * If we are attached to the DOM, simply switch the state of to the new value and  
+     * if there is a switch in state then the onChange method will be called, if necessary, by 
+     * the 'switchChange.bootstrapSwitch' callback.
+     * 
+     * If we are not attached to the DOM, manually call the onChange method to update the value 
+     * of the underlying {@link CheckBox} and fire change events. 
+     */
     @Override
     public void setValue(final Boolean value, final boolean fireEvents) {
-        Boolean oldValue = getValue();
         if (isAttached()) {
```

Comment:
Your proposed changes here cause don't work as expected.  **switchState(...) does not update the value of SimpleCheckBox**.  I am guessing this is a fundamental flaw in the ToggleSwitch implementation.  The onChange callback is called whenever the ToggleSwitch state is changed.  This is why we need to update the SimpleCheckBox's value when the ToggleSwitch state is changed.  Reverting the changes back to the original causes the original problem.